axum/src/routing/method_routing.rs

```diff
@@ -939,22 +936,89 @@ impl<ReqBody, E> MethodRouter<ReqBody, E> {
             mut allow_header,
             _request_body: _,
         } = self;
-        let svc = Some(Route::new(svc));
-        set_service!(
+
+        let svc = Route::new(svc);
+
+        set_service(
```

Comment:
These were somewhat more readable before. Generic closure would be a great solution, but we don't have those yet of course :/

Reply:
Yeah I agree its a bit unfortunate but this method was in the top 3 so think its worth it.